roles/launch_vms/files/upload_image

```diff
@@ -0,0 +1,159 @@
+#! /usr/bin/env python
+
+import os
+import ssl
+import time
+from urlparse import urlparse
+from httplib import HTTPSConnection
+
+from ovirtsdk4 import Connection, types
+
+MB_per_request = 32
+
+def main():
+    args = parse_args()
+    api = Connection(
+        url=args.url,
+        username=args.username,
+        password=args.password,
+        insecure=True
+    )
+
+    system_service = api.system_service()
+    transfers_service = system_service.image_transfers_service()
+    disks_service = system_service.disks_service()
+
+    disk = create_disk(disks_service, args.disk_name, args.disk_size,  args.storage_domain)
+    try:
+        wait_for_disk(disks_service, disk)
+        transfer_image(transfers_service, disk.id, args.image_path)
+        wait_for_disk(disks_service, disk)
+    except Exception as e:
+        if '409' not in ' '.join(e.args):
+            raise
+
+
+def get_resource(service, name):
+    try:
+        resource = service.list(search='name={}'.format(name))
+    except TypeError:
+        # Not all resource types allow the search parameter
+        # In this case, will have to manually filter
+        resource = list(filter(lambda x: x.name == name, service.list()))
+    return resource[0] if resource else None
+
+
+def create_disk(disks_service, disk_name, disk_size, storage_domain):
+    return get_resource(disks_service, disk_name) or disks_service.add(
+        types.Disk(
+            name=disk_name,
+            bootable=True,
+            active=True,
+            description='Template disk for creating rhel7 VMs',
+            format=types.DiskFormat.COW,
+            provisioned_size=disk_size * 2**30,
+            storage_domains=[
+                types.StorageDomain(
+                    name=storage_domain
+                )
+            ]
+        )
+    )
+
+
+def wait_for_disk(disks_service, disk):
+    disk_service = disks_service.disk_service(disk.id)
+    while disk.status != types.DiskStatus.OK:
+        time.sleep(5)
```

Comment:
every upload is going to be relatively long operation and our API is quite slow...15s sounds better to me